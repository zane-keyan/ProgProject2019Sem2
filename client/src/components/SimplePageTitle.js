import React, { Component } from "react";
import icon from "../images/icon-white.png";

class SimplePageTitle extends Component {
  showIcon = () => {
    if (this.props.doShowIcon) {
      return (
        <img
          style={styles().iconImgStyle}
          src={icon}
          className="img-thumbnail"
          alt="thumbnail"
        />
      );
    }
  };
  render() {
    return (
      <div className="title-container text-center text-light">
        <h1
          style={styles().titleStyle}
          className="text-light display-3 checkout-label"
        >
          {this.showIcon()}
          {this.props.title}
        </h1>
        <p style={styles().subtitleStyle} className="display-4 ">
          {this.props.subtitle}
        </p>
      </div>
    );
  }
}
var minStyles = {
  titleStyle: {
    fontSize: "3.5em"
  },
  iconImgStyle: {
    maxWidth: 80
  },
  list: {
    marginLeft: 0,
    paddingLeft: 0
  },
  subtitleStyle: {
    fontSize: "1.5em",
    margin: 5
  }
};
var maxStyles = {
  titleStyle: {
    // fontSize: "5em"
  },
  iconImgStyle: {
    maxWidth: 100
  },
  list: {}
};
var styles = function() {
  if (window.innerWidth <= 415) {
    return minStyles;
  }
  return maxStyles;
};

export default SimplePageTitle;
