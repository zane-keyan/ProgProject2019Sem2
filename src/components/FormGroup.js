import React, { Component } from "react";
class FormGroup extends Component {
  render() {
    return (
      <div class="form-group">
        <label className="font-weight-bold text-light form-label shadow-lg">
          {this.props.label}
        </label>
        <input
          type={this.props.inputType}
          name={this.props.name}
          id={this.props.id}
          placeholder={this.props.placeholder}
          class="form-control form-input text-light form-control-lg"
          
        />
      </div>
    );
  }
}

export default FormGroup;
