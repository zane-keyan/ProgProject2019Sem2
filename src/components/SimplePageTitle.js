import React, { Component } from "react";
import icon from "../images/icon-white.png";

class SimplePageTitle extends Component {
  showIcon = () => {
    if (this.props.doShowIcon) {
      return <img src={icon} class="img-thumbnail" />;
    }
  };
  render() {
    return (
      <div className="title-container text-center text-light">
        <h1 className="text-light display-3 checkout-label">
          {this.showIcon()}
          {this.props.title}
        </h1>
        <p className="display-4 ">{this.props.subtitle}</p>
      </div>
    );
  }
}

export default SimplePageTitle;
