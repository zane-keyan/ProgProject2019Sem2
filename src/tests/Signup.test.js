import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router, Link } from "react-router-dom";
import FormGroup from "../components/FormGroup";
import SimplePageTitle from "../components/SimplePageTitle";

it("render Signup page correctly", () => {
  const SignupPage = renderer
    .create(
      <Router>
        <Link to="/signup" />
      </Router>
    )
    .toJSON();
  expect(SignupPage).toMatchSnapshot();
});
it("render DateOfBirth FormGroup correctly", () => {
  const DateOfBirthFormGroup = renderer
    .create(
      <FormGroup
        inputType="date"
        label="Date of Birth"
        name="dateOfBirth"
        placeholder="dd/mm/yyyy"
      />
    )
    .toJSON();
  expect(DateOfBirthFormGroup).toMatchSnapshot();
});
it("render SimpleTitle  correctly", () => {
  const SimpleTitle = renderer
    .create(
      <SimplePageTitle
        title="Homy's car"
        subtitle="Sign up"
        doShowIcon={true}
      />
    )
    .toJSON();
  expect(SimpleTitle).toMatchSnapshot();
});
