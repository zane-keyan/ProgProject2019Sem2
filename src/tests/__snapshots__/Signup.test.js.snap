// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DateOfBirth FormGroup correctly 1`] = `
<div
  class="form-group"
>
  <label
    className="font-weight-bold text-light form-label shadow-lg"
  >
    Date of Birth
  </label>
  <input
    class="form-control form-input text-light form-control-lg"
    name="dateOfBirth"
    placeholder="dd/mm/yyyy"
    type="date"
  />
</div>
`;

exports[`render Signup page correctly 1`] = `
<a
  href="/signup"
  onClick={[Function]}
/>
`;

exports[`render SimpleTitle  correctly 1`] = `
<div
  className="title-container text-center text-light"
>
  <h1
    className="text-light display-3 checkout-label"
  >
    <img
      alt="thumbnail"
      class="img-thumbnail"
      src="icon-white.png"
    />
    Homy's car
  </h1>
  <p
    className="display-4 "
  >
    Sign up
  </p>
</div>
`;
